projects/material-addons/src/lib/numeric-field/numeric-field.directive.ts

```diff
@@ -260,14 +260,9 @@ export class NumericFieldDirective implements OnInit, OnDestroy, AfterViewChecke
     if (!!this.unit && !this.unitSpan && this.inputEl.nativeElement.offsetWidth !== 0) {
       // Get the input wrapper and apply necessary styles
       const inputWrapper = this.inputEl.nativeElement.parentNode.parentNode;
-      // this.renderer.addClass(inputWrapper, 'numeric-input-wrapper');
 
       // Create the span with unit symbol and apply necessary styles
-      if (!this.unitSpan) {
-        this.unitSpan = this.renderer.createElement('span');
-        const unitSymbol = this.renderer.createText(this.unit);
-        this.renderer.appendChild(this.unitSpan, unitSymbol);
-      }
+      this.unitSpan = this.renderer.createElement('span');
```

Comment:
Ich bin mir nicht sicher was die Idee hinter dem ursprünglichen If war, ich hoffe wir erzeugen nun nicht endlos viele Spans.

Reply:
was das if da vorher sollte weiß ich auch nicht, die Bedingung  ist aber bereits in Zeile 260 auch schon drinnen!

Reply:
Aja, das stimmt.